pkg/sql/gcjob_test/gcjob_test_test: TestSchemaChangeGCJob failed

pkg/sql/gcjob_test/gcjob_test_test.TestSchemaChangeGCJob [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/13445744?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/13445744?buildTab=artifacts#/) on master @ [58502d1125b70dafeea34b96733967a387f9ef0d](https://github.com/cockroachdb/cockroach/commits/58502d1125b70dafeea34b96733967a387f9ef0d):


```
=== RUN   TestSchemaChangeGCJob
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/4d64db87b2aa0e9fccbce189bc4b8745/logTestSchemaChangeGCJob292038267
    test_log_scope.go:81: use -show-logs to present logs inline
    gc_job_test.go:85: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4d64db87b2aa0e9fccbce189bc4b8745/logTestSchemaChangeGCJob292038267
--- FAIL: TestSchemaChangeGCJob (77.99s)
=== RUN   TestSchemaChangeGCJob/dropItem=0/ttlTime=0
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    conditional_wrap.go:190: 
        pkg/sql/gcjob_test/gcjob_test_test_test/pkg/sql/gcjob_test/gc_job_test.go:112: (doTestSchemaChangeGCJob)
        	NOTICE: .JobRegistry() called via implicit interface ApplicationLayerInterface;
        HINT: consider using .ApplicationLayer().JobRegistry() instead.
    conditional_wrap.go:194: TIP: consider replacing the test server initialization from:
            ts, ... := serverutils.StartServer(t, ...)
            defer ts.Stopper().Stop(...)
        to:
            srv, ... := serverutils.StartServer(t, ...)
            defer srv.Stopper().Stop(...)
            ts := srv.ApplicationLayer()
        
        See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
    gc_job_test.go:236: condition failed to evaluate within 45s: from jobs_verification.go:178: job 0: expected running status waiting for GC TTL, got deleting data
    --- FAIL: TestSchemaChangeGCJob/dropItem=0/ttlTime=0 (47.81s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSchemaChangeGCJob.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35209

Also see here:
https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/13706182?buildTab=tests&expandedTest=build%3A%28id%3A13706161%29%2Cid%3A140358

```
    gc_job_test.go:236: condition failed to evaluate within 45s: from jobs_verification.go:178: job 0: expected running status waiting for GC TTL, got deleting data
    --- FAIL: TestSchemaChangeGCJob/dropItem=1/ttlTime=1 (47.74s)
```

Seems to still be flaky

--- a/pkg/sql/gcjob_test/gc_job_test.go
+++ b/pkg/sql/gcjob_test/gc_job_test.go
@@ -234,9 +234,18 @@ func doTestSchemaChangeGCJob(t *testing.T, dropItem DropItem, ttlTime TTLTime) {
 	// Check that the job started.
 	jobIDStr := strconv.Itoa(int(job.ID()))
 	testutils.SucceedsSoon(t, func() error {
-		return jobutils.VerifyRunningSystemJob(
+		if err := jobutils.VerifyRunningSystemJob(
 			t, sqlDB, 0, jobspb.TypeSchemaChangeGC, sql.RunningStatusWaitingGC, lookupJR,
-		)
+		); err != nil {
+			// Since the intervals are set very low, the GC TTL job may have already
+			// started. If so, the status will be "deleting data" since "waiting for
+			// GC TTL" will have completed already.
+			if testutils.IsError(err, "expected running status waiting for GC TTL, got deleting data") {
+				return nil
+			}
+			return err
+		}
+		return nil
 	})
 
 	if ttlTime != FUTURE {

